packages/light-client-extension-helpers/src/background/background-helper.ts

```diff
@@ -619,50 +628,47 @@ const withClient =
   }
 
 const getChainData = withClient(async (client) => {
-  const [genesisHash, name] = await Promise.all(
-    ["chainSpec_v1_genesisHash", "chainSpec_v1_chainName"].map((method) =>
-      substrateClientRequest<string>(client, method),
-    ),
-  )
+  const [genesisHash, name, { ss58Format }] = (await Promise.all(
+    [
+      "chainSpec_v1_genesisHash",
+      "chainSpec_v1_chainName",
+      "chainSpec_v1_properties",
+    ].map((method) => substrateClientRequest(client, method)),
+  )) as [string, string, { ss58Format: number }]
   return {
     genesisHash,
     name,
+    ss58Format,
   }
 })
 
-const getSs58Format = withClient(
+// TODO: update this implementation when these issues are implemented
+// https://github.com/paritytech/json-rpc-interface-spec/issues/110
+// https://github.com/smol-dot/smoldot/issues/1186
+const getGenesisStateRoot = withClient(
```

Comment:
@josepot I'm thinking about replacing this implementation with
```ts
const getGenesisStateRoot2 = withClient(async (client) => {
  const { unfollow, follow$ } = getObservableClient(client).chainHead$()
  const stateRoot = await firstValueFrom(
    follow$.pipe(
      mergeMap(async (event) => {
        const { stateRoot } = await substrateClientRequest<{
          stateRoot: string
        }>(client, "chain_getHeader", [
          await substrateClientRequest<string>(
            client,
            "chainSpec_v1_genesisHash",
          ),
        ])
        return stateRoot
      }),
    ),
  )
  unfollow()
  return stateRoot
})
```

Questions
- Does it make sense to add the `/client` and `rxjs` dependencies to the helpers for this?
- the `unfollow()`, Is it redundant in this context? I'm not able to find if the `follow$` observable does the `unfollow` when there are no more subscribers.
- `follow$` doesn't seem to do any unpinning, right?

Reply:
As discussed offline, all good with this:

```ts
const getGenesisStateRoot2 = withClient(async (client) => {
  const chainHead = getObservableClient(client).chainHead$()
  await firstValueFrom(chainHead.runtime$)
  const genesisHash = await substrateClientRequest<string>(
    client,
    "chainSpec_v1_genesisHash",
  )
  const { stateRoot } = await substrateClientRequest<{
    stateRoot: string
  }>(client, "chain_getHeader", [genesisHash])
  chainHead.unfollow()
  return stateRoot
})
```